warden/x/async/keeper/abci.go

```diff
@@ -51,8 +73,27 @@ func (k Keeper) ExtendVoteHandler() sdk.ExtendVoteHandler {
 			}
 		}
 
+		pVotes, done := k.p.Votes()
+		defer done()
+
+		mapVote := func(err error) types.FutureVoteType {
+			if err == nil {
+				return types.FutureVoteType_VOTE_TYPE_VERIFIED
+			}
+			return types.FutureVoteType_VOTE_TYPE_REJECTED
+		}
+
+		votes := make([]*types.VEVoteItem, len(pVotes))
+		for i, v := range pVotes {
+			votes[i] = &types.VEVoteItem{
+				FutureId: v.ID,
+				Vote:     mapVote(v.Err),
+			}
+		}
```

Comment:
let's avoid the overhead of a closure, and simplify the code:

```suggestion
		votes := make([]*types.VEVoteItem, len(pVotes))
		for i, v := range pVotes {
			vote := types.FutureVoteType_VOTE_TYPE_VERIFIED
			if v.Err != nil {
				vote = types.FutureVoteType_VOTE_TYPE_REJECTED
		    }
			votes[i] = &types.VEVoteItem{
				FutureId: v.ID,
				Vote:     vote,
			}
		}
```

Reply:
But here's no closure introduced, there would be a closure if I used any variable outside of outer context

Reply:
Changed

Reply:
afaik the Go compiler doesn't really make distinctions between anonymous functions and closures so it's effectively the same :)

in any case, the idiomatic way of doing it in Go is directly in the loop